src/xsl/openrosa2html5form.xsl

```diff
@@ -1394,10 +1333,19 @@ XSLT Stylesheet that transforms OpenRosa style (X)Forms into valid HTMl5 forms
     </xsl:template>
 
     <xsl:template name="text-content">
-        <xsl:if test="string-length(.) = 0">
-            <xsl:text><!-- avoids self-closing tags on empty elements --> </xsl:text>
-        </xsl:if>
-        <xsl:apply-templates /><!-- call xf:output template if output is present -->
+        <xsl:choose>
+            <xsl:when test="string-length(.) = 0">
+                <xsl:text><!-- avoids self-closing tags on empty elements --> </xsl:text>
+                <xsl:apply-templates />
+            </xsl:when>
+            <!-- Firefox does not apply templates to these attributes -->
+            <xsl:when test="name() = 'jr:constraintMsg' or name() = 'jr:requiredMsg' or ($openclinica = 1 and starts-with(name(), 'oc:') and substring(name(), string-length(name()) - string-length('Msg') + 1) = 'Msg' )">
+                <xsl:value-of select="." />
+            </xsl:when>
+            <xsl:otherwise>
+                <xsl:apply-templates />
+            </xsl:otherwise>
+        </xsl:choose>
     </xsl:template>
 
     <xsl:template name="translations">
```

Comment:
GitHub doesn't let me add a comment in the collapsed bit below, but a note for future performance improvement, a few lines below:

```xsl
<xsl:for-each select="/h:html/h:head/xf:model/xf:itext/xf:translation/xf:text[@id=$id]">
```

This specific `for-each` and its XPath expression accounts for **nearly all** of the WHO VA form's transform time. Not the logic within the `for-each`, it takes nearly as much time with that logic removed.

I considered also moving this out to `transformer.ts`, but it's out of scope and of course this PR is already **enormous**.